spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/TaskDeploymentIdNotAvailableException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.repository;
+
+/**
+ * Thrown when the task ID is not returned during task launch.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class TaskDeploymentIdNotAvailableException extends RuntimeException {
```

Comment:
I think `IllegalStateException` would be fine, this is unlikely to happen unless the SPI implementation is not doing what it is supposed to